scalikejdbc-core/src/main/scala/scalikejdbc/ParameterBinderFactory.scala

```diff
@@ -0,0 +1,111 @@
+package scalikejdbc
+
+import java.io.InputStream
+import java.sql.PreparedStatement
+import scalikejdbc.UnixTimeInMillisConverterImplicits._
+import scalikejdbc.interpolation.SQLSyntax
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox.Context
+
+private[scalikejdbc] case class SQLSyntaxParameterBinder(syntax: SQLSyntax) extends ParameterBinder[SQLSyntax] {
+  val value = syntax
+  def apply(stmt: PreparedStatement, idx: Int): Unit = ()
+}
+
+trait ParameterBinderFactory[A] { self =>
+
+  def apply(value: A): ParameterBinder[A]
+
+  def bimap[B](f: A => B, g: B => A): ParameterBinderFactory[B] = new ParameterBinderFactory[B] {
```

Comment:
you're right!

in cats, `imap` https://github.com/typelevel/cats/blob/b638e298ca544739602e76388c9d266c4bc4357d/core/src/main/scala/cats/functor/Invariant.scala#L10

Should we choose them?